Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm

```diff
@@ -152,6 +199,45 @@ void initializeWebViewConfiguration(const char* libraryPath, WKStringRef injecte
         reinterpret_cast<IMP>(swizzledProcessImageAnalysisRequest)
     );
 #endif
+
+#if PLATFORM(MAC)
+    // See NetworkProcess::platformInitializeNetworkProcessCocoa for supporting a local DNS resolver on iOS Simulator.
+
+    if (!options.useLocalDNSResolver)
+        return;
+
+    NSUUID *agentUUID = [[NSUUID alloc] init];
+    if (!agentUUID) {
```

Comment:
Can this really happen? The document doesn't mention it.

Reply:
Maybe not. I see similar checks in other code bases, but it's not consistent - and existing usage in WebKit doesn't check the result. I'll remove it.